engine/ekf.py

```diff
@@ -101,14 +103,61 @@ def get_h_jac(self, pose):
         """
         return np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
 
-    def predict_step(self, control):
+    def get_arc_lengths(self, left_side_w, right_side_w):
+        """
+        Returns the arc_lengths of the left side of the robot and the right side of the robot 
+        given the angular velocity of the left side and angular velocity of the right side. 
+
+        left_side_w: angular velocity of the motors on the left side of the robot
+        right_side_w: angular velocity of the motors on the right side of the robot
+        """
+        # TODO: from electrical: from motors, get angular velocity
+        # TODO: from angular velocity, find arc_lengths 
+        # Note: this was put on hold Spring '23 because Professor advised that getting velocity
+        # in this fashion was not that accurate 
+        left_arc_length = left_side_w # TEMP
+        right_arc_length = right_side_w # TEMP
+
+        return [left_arc_length, right_arc_length]
+
+
+    def get_controls(self, arc_lengths): 
+        """
+        Returns phi, the angle changed in a timestep and d, the displacement traveled in a timestep.
+
+        Given from electrical: arc_lengths: a tuple such that the first entry is the 
+        left-side arc distance and the second entry is the right-side arc distance
+        
+        Arc distance is calculated from motor_velocity * timestep
+        """
+
+        L_L = arc_lengths[0]
+        L_R = arc_lengths[1]
+
+        try:
```

Comment:
What if L_R = L_L?

Reply:
if L_R = L_L, i set r = 0 to avoid division by 0 - since r is the distance from the robot to center of curvature, i think 0 would make sense

Reply:
sg

Reply:
yeah the name was shortened on the documentation cuz it would be annoying to add into all the equations but i can update this